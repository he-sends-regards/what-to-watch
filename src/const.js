export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  MY_LIST: `/mylist`,
  ADD_REVIEW: `/films/:id/review`,
  FILM: `/films`,
  PLAYER: `/player`
};

export const APIRoute = {
  FILMS: `/films`,
  LOGIN: `/login`,
  PROMO_FILM: `/films/promo`,
  REVIEWS: `/comments`,
  FAVORITE: `/favorite`
};

export const HttpCode = {
  UNAUTHORIZED: 401
};

export const FilmRateDescriptions = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
  UNEXPECTED_RATE: `Unexpected rate`
};

export const FilmDescriptionTabsNames = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const ALL_GENRE = `All genres`;

export const FavoriteFlags = {
  ADD_TO_FAVORITE: 1,
  REMOVE_FROM_FAVORITE: 0
};
