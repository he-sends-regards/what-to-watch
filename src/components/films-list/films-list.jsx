import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import FilmCard from "../film-card/film-card";
import {filmValidator} from "../../propsValidators/validators.prop";
import {withFilmsList} from "../../hocs/with-films-list/with-films-list";
import {ALL_GENRE} from "../../const";

const MORE_LIKE_THIS_FILMS_COUNT = 4;

const FilmsList = (props) => {
  const {activeFilmId, onMouseEnter, onMouseLeave, showFilmsCount, films, doesBelongMoreLikeThis} = props;

  return (
    <div className="catalog__movies-list">
      {
        (doesBelongMoreLikeThis
          ? films
            .filter((film) => {
              return film.id !== props.currentFilmId && film.genre === props.currentFilmGenre;
            })
            .slice(0, MORE_LIKE_THIS_FILMS_COUNT)
          : films
            .filter((film) => props.activeGenre === ALL_GENRE || props.activeGenre === film.genre)
            .slice(0, showFilmsCount)
        ).map((film) => {
          return (
            <FilmCard
              key={film.id}
              film={film}
              isActive={film.id === activeFilmId}
              onMouseEnter={onMouseEnter}
              onMouseLeave={onMouseLeave}
            />
          );
        })
      }
    </div>
  );
};

FilmsList.propTypes = {
  films: PropTypes.arrayOf(PropTypes.shape(filmValidator)),
  showFilmsCount: PropTypes.number.isRequired,
  activeFilmId: PropTypes.number.isRequired,
  onMouseEnter: PropTypes.func.isRequired,
  onMouseLeave: PropTypes.func.isRequired,
  doesBelongMoreLikeThis: PropTypes.bool.isRequired,
  currentFilmId: PropTypes.oneOfType([PropTypes.number.isRequired, PropTypes.oneOf([undefined]).isRequired]),
  currentFilmGenre: PropTypes.oneOfType([PropTypes.string.isRequired, PropTypes.oneOf([undefined]).isRequired]),
  activeGenre: PropTypes.string.isRequired
};

const mapStateToProps = ({FILMS}) => ({
  showFilmsCount: FILMS.showFilmsCount,
  activeGenre: FILMS.activeGenre
});

export {FilmsList};
export default connect(mapStateToProps)(withFilmsList(FilmsList));
