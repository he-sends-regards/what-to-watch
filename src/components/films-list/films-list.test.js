import React from "react";
import renderer from "react-test-renderer";
import {FilmsList} from "./films-list";
import mockFilms from "../../mocks/films";
import {ALL_GENRE} from "../../const";

describe(`Should FilmsList render correctly`, () => {
  const noop = () => {};
  const filmsListComponent = renderer.create(
      <FilmsList
        activeFilmId={1}
        onMouseEnter={noop}
        onMouseLeave={noop}
        activeGenre={ALL_GENRE}
        showFilmsCount={2}
        films={mockFilms}
        doesBelongMoreLikeThis={true}
      />
  );
  it(`Should filmsListComponent render correctly`, () => {
    expect(filmsListComponent.toJSON()).toMatchSnapshot();
  });
});
