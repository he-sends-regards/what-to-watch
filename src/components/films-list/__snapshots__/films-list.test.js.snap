// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FilmsList render correctly Should filmsListComponent render correctly 1`] = `
<div
  className="catalog__movies-list"
/>
`;
